File: mgyoutube-multitech/api-python/repos/UserDataRepo.py
from abc import ABC, abstractmethod
from apimodel.User import User


class UserDataRepo(ABC):

    @abstractmethod
    def repositoryStartup(self):
        pass

    @abstractmethod
    def getUserByUsername(self,  username: str) -> User:
        pass

    @abstractmethod
    def addUser(self,  user: User) -> User:
        pass

    @abstractmethod
    def removeUser(self,  user: User):
        pass

    @abstractmethod
    def replaceUser(self, userId: str, user: User) -> User:
        pass

    @abstractmethod
    def addChildToParent(self, parentUserId: str, childUserId: str):
        pass

    @abstractmethod
    def getChildrenForParent(self, userId: str) -> list:
        pass

File: mgyoutube-multitech/api-node/src/repos/userdatarepo.ts
import { User } from '../apimodel/user';


export interface UserDataRepo {

    repositoryStartup(): Promise<void>;

    getUserByUsername(username: string): Promise<User | undefined>;

    addUser(user: User): Promise<User | undefined>;

    // updateUser(userId, user) {
    //     return this.userDataRepo.replaceUser(userId, user);
    // }


    removeUser(user: User): Promise<void>;

    replaceUser(userId: string, user: User): Promise<User | undefined>;


    addChildToParent(parentUserId: string, childUserId: string): Promise<void>;


    getChildrenForParent(parentUserId: string): Promise<Array<User>>;

}
